### ISSUE TITLE
Broken exports

### ISSUE THREAD

**Issue by RobinTail** [thebuilder/react-intersection-observer#694]
@thebuilder , something changed in between 9.10.3 and 9.12.0 that broke the existing imports:

Getting the following error for all my tests:
```
TS2307: Cannot find module 'react-intersection-observer/test-utils' or its corresponding type declarations.
```

<img width="1144" alt="Screenshot 2024-07-11 at 10 40 00" src="https://github.com/thebuilder/react-intersection-observer/assets/13189514/d59f92b7-99ea-4e0d-b56f-55522d8a8310">

**Comment by RobinTail** [comment]
Narrowed it down to 9.11.0

**Comment by RobinTail** [comment]
Some issues revealed using `attw`:

https://arethetypeswrong.github.io/?p=react-intersection-observer%409.11.0

<img width="691" alt="Screenshot 2024-07-11 at 10 48 32" src="https://github.com/thebuilder/react-intersection-observer/assets/13189514/335f14c6-39bd-4790-b997-92b52bc8a87f">

**Comment by RobinTail** [comment]
https://arethetypeswrong.github.io/?p=react-intersection-observer%409.11.0
@thebuilder 


| 9.10.3 | 9.11.0 |
|--------|--------|
| <img width="684" alt="Screenshot 2024-07-11 at 10 58 36" src="https://github.com/thebuilder/react-intersection-observer/assets/13189514/b64bbf5e-42ed-4045-b8f4-eee69535a624"> | 
<img width="680" alt="Screenshot 2024-07-11 at 10 59 17" src="https://github.com/thebuilder/react-intersection-observer/assets/13189514/753118d0-6789-45b5-8124-bdb8de5775a4"> |

**Comment by RobinTail** [comment]
In order to fix it, `tsup` should be configured to build test utils as `index.js` inside the `test-utils` folder (along with "dist")

**Comment by RobinTail** [comment]
Made a fix. I had a similar issue in my project.

**Comment by thebuilder** [comment]
Thanks a lot - ESM is the bane of my existence.

**Comment by RobinTail** [comment]
I confirm all issues fixed in 9.13.0, including #689
